packages/rocketchat-api/server/v1/rooms.js

```diff
@@ -115,3 +115,22 @@ RocketChat.API.v1.addRoute('rooms.saveNotification', { authRequired: true }, {
 		return RocketChat.API.v1.success();
 	}
 });
+
+RocketChat.API.v1.addRoute('rooms.favorite', { authRequired: true }, {
+	post() {
+		const { favorite, roomId } = this.bodyParams;
```

Comment:
Let's make it so we also accept roomName like we do for channels. Like this method but allow it to accept any type of room and for features such as favorite, ensure they are part of that room (have a subscription). Especially since the `toggleFavorite` method doesn't check to ensure they have a subscription or are part of that room.

https://github.com/RocketChat/Rocket.Chat/blob/a9aea3640262482d7111abe3b5c231aa80da897e/packages/rocketchat-api/server/v1/channels.js#L3-L30